spec/PeriodicBackgroundSync-index.bs

```diff
@@ -59,23 +70,222 @@ Introduction {#intro}
+    1. Set |delayForOrigin| to the multiple of |effectiveGap| greater than or equal to [=options/minInterval=].
+    1. Let |timeTillScheduledEventForOrigin|, a number, be the time till the next scheduled [=periodicSync event=] for |origin|, if any, null otherwise.
+    1. If |timeTillScheduledEventForOrigin| is not null:
+        1. If |timeTillScheduledEventForOrigin| - |delayForOrigin| is greater than or equal to |minIntervalForOrigin|, abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| and abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|.
+    1. Let |timeSinceLastPeriodicSync| be null if [=time the last periodicSync event was fired=] is null, else |now| - [=time the last periodicSync event was fired=].
+    1. If |timeSinceLastPeriodicSync| is null, set |timeToFire| to |delayForOrigin| + |now|.
+    1. Else:
+        1. If |timeSinceLastPeriodicSync| is greater than equal to [=minimum interval across origins=], set |timeToFire| to |delayForOrigin| + |now|.
+        1. Else, set |timeTillNextAllowedPeriodicSync| to [=minimum interval across origins=] - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
+    1. Set the [=periodicsync registration/time to fire=] of |registration| to |timeToFire|.
+1. Else:
+    1. Increment [=periodicsync registration/count of retries=].
+    1. If [=periodicsync registration/count of retries=] is greater than the [=maximum number of retries=] allowed, set [=periodicsync registration/count of retries=] to 0 and follow the steps to [=calculate the time to fire of the first attempt of=] the periodicSync event.
+    1. Else, Set the [=periodicsync registration/time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to [=periodicsync registration/count of retries=].
+
+## <dfn>Schedule processing</dfn> ## {#schedule-delayed-processing}
+To [=schedule processing=] of a [=periodicSync registration=] |registration|, run the following steps:
+1. Assert: |registration|'s associated [=periodicsync registration/registration state=] is [=pending=].
+1. [=In parallel=], schedule a task to [=fire a periodicSync event=] for |registration| as soon as the the device is online at or after |registration|'s [=periodicsync registration/time to fire=].
```

Comment:
Is there a recommended construct I can use here?

Reply:
I wonder if this is too much detail, because it opens questions like "what if the browser restarts" and has overlap with https://www.w3.org/TR/task-scheduler/. Maybe going a step more abstract is the right way to go, and phrasing this as "schedule `firing a periodicSync event` as soon as the..." is an idea?